AmplifyPlugins/Auth/Tests/AWSCognitoAuthPluginUnitTests/Support/CustomEndpoint/EndpointResolving+ValidationStepTestCase.swift

```diff
@@ -0,0 +1,145 @@
+                XCTAssertEqual(
+                    e?.recoverySuggestion,
+                    """
+                    Invalid scheme for value `endpoint`: \(invalidInput).
+                    AWSCognitoAuthPlugin only supports the https scheme.
+                    > Remove the scheme in your `endpoint` value.
+                    e.g.
+                    "endpoint": \(URL(string: invalidInput)?.host ?? "example.com")
+                    """
+                )
+            }
+        }
+    }
+
+    // MARK: EndpointResolving.ValidationStep.validURL()
+    /// Given: A String representation of a URL.
+    /// When: That String contains a valid URL.
+    /// Then: ValidationStep should not throw.
+    func testValidURL_Valid() throws {
+        let validInput = "foo.com"
+        XCTAssertNoThrow(
+            try EndpointResolving.ValidationStep.validURL()
+                .validate(validInput)
+        )
+    }
+
+    /// Given: A String representation of a URL.
+    /// When: That String contains an invalid URL.
+    /// Then: ValidationStep should throw an error with expected output.
+    func testValidURL_Invalid() throws {
+        let invalidInput = "\\"
+        XCTAssertThrowsError(
+            try EndpointResolving.ValidationStep.validURL()
+                .validate(invalidInput),
+            ""
+        ) { error in
+            let e = error as? AuthError
+            XCTAssertEqual(
+                e?.errorDescription,
+                "Error configuring AWSCognitoAuthPlugin"
+            )
+            XCTAssertEqual(
+                e?.recoverySuggestion,
+                """
+                Invalid value for `endpoint`: \(invalidInput)
+                Expected valid url, received: \(invalidInput)
+                > Replace \(invalidInput) with a valid URL.
+                """
+            )
+        }
+    }
+
+    // MARK: EndpointResolving.ValidationStep.pathIsEmpty()
+    /// Given: A String representation of a URL.
+    /// When: That String doesn't contain a path.
+    /// Then: ValidationStep should not throw.
+    func testPathIsEmpty_Valid() throws {
+        let validInput = "https://foo.com"
+        let components = URLComponents(string: validInput)!
+
+        XCTAssertNoThrow(
+            try EndpointResolving.ValidationStep.pathIsEmpty()
+                .validate((components, validInput))
+        )
+    }
+
+    /// Given: A String representation of a URL.
+    /// When: That String contains a path.
+    /// Then: ValidationStep should throw an error with expected output.
+    func testPathIsEmpty_Invalid() throws {
+        let invalidInput = "https://foo.com/hello"
+        let components = URLComponents(string: invalidInput)!
+
+        XCTAssertThrowsError(
+            try EndpointResolving.ValidationStep.pathIsEmpty()
+                .validate((components, invalidInput)),
+            ""
+        ) { error in
+            let e = error as? AuthError
+            XCTAssertEqual(
```

Comment:
Instead of checking the error description can we just check if the error is of type `AuthError.configuration`. The description can change so better to check the type?

Reply:
Good call. Addressed in [3fab560](https://github.com/aws-amplify/amplify-ios/pull/2060/commits/3fab5609034d61be7051422aadb7c388e86755ba).